### PR TITLE
test: use earthly-next branch for some tests

### PR DESCRIPTION
**Pull request by alexcb**
Adds a new workflow to run the +docker-tests-no-qemu-group* tests using the buildkit version from the earthly-next branch.

### CODE REVIEW

**Comment by alexcb** [comment]
image

**Comment by alexcb** [comment]
here's the new `earthly-next-docker-tests` test group:
image

I left off the podman tests.

File: .github/workflows/ci-earthly-next-docker-ubuntu.yml [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  # this job will output a boolean value to check whether files that require these tests to run
+  # since all jobs depend on `build-earthly` job, conditionally running it will either cause all jobs to run or skip,
+  # depending on the output value of this job
+#  check-essential-files:
```
Comment (mikejholly) [inline .github/workflows/ci-earthly-next-docker-ubuntu.yml:23]: Do we need to retain this?

File: .github/workflows/reusable-test.yml [hunk 2]
```diff
@@ -78,6 +83,12 @@ jobs:
       - name: Execute ${{ inputs.TEST_TARGET }} (Earthly Only)
         run: |-
           ${{inputs.SUDO}} ${{ inputs.BUILT_EARTHLY_PATH }} ${{inputs.EXTRA_ARGS}} --ci -P ${{inputs.TEST_TARGET}}
+      - name: Display buildkit version
+        run: |-
+          ${{inputs.SUDO}} ${{inputs.BINARY}} ps -a
+          ${{inputs.SUDO}} ${{inputs.BINARY}} logs earthly-buildkitd |& grep 'starting earthly-buildkit'
```
[the remark this answers is not in the document]
Comment (mikejholly) [reply .github/workflows/reusable-test.yml:89]: Works for me!

```
$ echo hi |& cat
hi
```

File: .github/workflows/ci-earthly-next-docker-ubuntu.yml (hunk 1 above)
Comment (alexcb) [reply .github/workflows/ci-earthly-next-docker-ubuntu.yml:23]: good point; it was copy/pasted but I don't think it applies to us. I'll remove it.

**Pull request merged**